From bb9b48b7b5234df280a4cd6d72ca1ea9e83ebb35 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Thu, 17 Oct 2024 22:45:51 +0800
Subject: [PATCH 100/104] AOSCOS: tty: mips_ejtag_fdc: Fix another passing
 incompatible pointer type warning
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

mips_ejtag_fdc_encode() method was changed to accept u8** since
ce7cbd9a6c81, but not all occurrences were changed to adapt it.

Some of the warnings or errors was fixed in a previous commit, this
commit fixes one more of them.

drivers/tty/mips_ejtag_fdc.c: In function ‘kgdbfdc_push_one’:
drivers/tty/mips_ejtag_fdc.c:1224:38: error: passing argument 1 of ‘mips_ejtag_fdc_encode’ from incompatible pointer type [-Werror=incompatible-pointer-types]
 1224 |         word = mips_ejtag_fdc_encode(bufs, &kgdbfdc_wbuflen, 1);
      |                                      ^~~~
      |                                      |
      |                                      const char **
drivers/tty/mips_ejtag_fdc.c:216:57: note: expected ‘const u8 **’ {aka ‘const unsigned char **’} but argument is of type ‘const char **’
  216 | static struct fdc_word mips_ejtag_fdc_encode(const u8 **ptrs,
      |                                              ~~~~~~~~~~~^~~~

Fixes: ce7cbd9a6c81 ("tty: mips_ejtag_fdc: use u8 for character pointers")
Fixes: 188942f05ce4 ("tty: mips_ejtag_fdc: Fix passing incompatible pointer type warning")
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 drivers/tty/mips_ejtag_fdc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/tty/mips_ejtag_fdc.c b/drivers/tty/mips_ejtag_fdc.c
index afbf7738c7c4..1cc775e74f92 100644
--- a/drivers/tty/mips_ejtag_fdc.c
+++ b/drivers/tty/mips_ejtag_fdc.c
@@ -1215,7 +1215,7 @@ static int kgdbfdc_read_char(void)
 /* push an FDC word from write buffer to TX FIFO */
 static void kgdbfdc_push_one(void)
 {
-	const char *bufs[1] = { kgdbfdc_wbuf };
+	const u8 *bufs[1] = { kgdbfdc_wbuf };
 	struct fdc_word word;
 	void __iomem *regs;
 	unsigned int i;
-- 
2.47.0

